src/network/network.cpp

```diff
@@ -708,7 +708,7 @@ void NetworkClientConnectGame(NetworkAddress address, CompanyID join_as, const c
 	_network_join_status = NETWORK_JOIN_STATUS_CONNECTING;
 	ShowJoinStatusWindow();
 
-	new TCPClientConnecter(address);
+	new TCPClientConnecter(NetworkAddress(hostname, port));
```

Comment:
Why not "collapse" this constructor too, so that it just takes hostname & port directly?

Reply:
There are at least two other places where TCPClientConnecter is used that pass NetworkAddress references, so it is not worth the effort here and the main goal was to remove "access" to os_abstraction.h from as much of the non-network code as possible due to the addition of safeguards for not using errno/strerror.